ENH: More complete raw reading

Eric89GXL opened the pull request:
Fixes / adds a bunch of fields we didn't have for Raw. I also finally got rid of the `read/write_extra_meas_info` functions, because the fields are now read/written manually. I didn't add any explicit tests because the I/O is implicitly tested already.

Eric89GXL commented on mne/io/fiff/raw.py:
```diff
@@ -158,6 +158,23 @@ def _preload_data(self, preload):
+        If data were already preloaded, it will do nothing.
+        """
+        if not self.preload:
+            self._preload_data(True)
```
I've wanted to be able to do this before. Now we can...

agramfort replied:
fine with me. I'd document this in API section in what's new though

agramfort commented on mne/io/constants.py:
```diff
@@ -747,18 +750,7 @@ def __init__(self, **kwargs):
 
-FIFF.FIFFV_POINT_LPA      = 1
-FIFF.FIFFV_POINT_NASION   = 2
-FIFF.FIFFV_POINT_RPA      = 3
```
why removing these?

Eric89GXL replied:
they are already defined elsewhere in the file

agramfort replied:
ok

agramfort commented:
travis is not happy

that's it for me

Eric89GXL commented:
Fixed tests, updated `whats_new.py`, and unified some `Info` handling for our non-FIFF formats. Tests pass here and on Travis/AppVeyor. Ready for review/merge from my end.

agramfort commented on mne/io/meas_info.py:
```diff
@@ -693,11 +781,75 @@ def write_meas_info(fid, info, data_type=None, reset_range=True):
+            write_int(fid, FIFF.FIFF_HPI_NCOIL, hpi_meas['ncoil'])
+        if hpi_meas.get('first_samp') is not None:
+            write_int(fid, FIFF.FIFF_FIRST_SAMPLE, hpi_meas['first_samp'])
+        if hpi_meas.get('lastmne/viz/tests/test_montage.py_samp') is not None:
```
why this?

Eric89GXL replied:
Whoa, brutal typo

agramfort commented:
besides LGTM

can somebody else have a look?

Eric89GXL commented:
Typo fixed.

mainakjas commented on doc/source/whats_new.rst:
```diff
@@ -111,6 +111,8 @@ API
 
    - When computing the noise covariance or MNE inverse solutions, the rank is estimated empirically using more sensitive thresholds, which stabilizes results by `Denis Engemann`_ and `Eric Larson`_ and `Alex Gramfort`_
 
+   - Raw FIFF files can be preloaded after class istantiation using ``raw.preload_data()``.
```
instantiation

agramfort commented:
thanks @Eric89GXL 

I'll fix typo in master
